openmc/universe.py

```diff
@@ -76,10 +76,17 @@ def __eq__(self, other):
             return False
         elif self.name != other.name:
             return False
-        elif self.cells != other.cells:
+        elif len(self.cells) != len(other.cells):
             return False
-        else:
-            return True
+
+        # Check each cell
+        for cell_id in self.cells:
+            if cell_id not in other.cells:
+                return False
+            if self.cells[cell_id] != other.cells[cell_id]:
+                return False
```

Comment:
Yes, your much simpler approach does indeed work, thanks! Do note that in the process, I also discovered bugs in the `Lattice.__eq__(...)` method - namely, when comparing the `pitch`, `lower_left` and `universes` attribute collections - and fixed them with the `numpy.any()` method.